qemu_v8.mk

```diff
@@ -16,7 +16,7 @@ include common.mk
 ARM_TF_PATH		?= $(ROOT)/arm-trusted-firmware
 
 EDK2_PATH		?= $(ROOT)/edk2
-EDK2_BIN		?= $(EDK2_PATH)/Build/ArmVirtQemuKernel-AARCH64/DEBUG_GCC49/FV/QEMU_EFI.fd
+EDK2_BIN		?= $(ROOT)/Build/ArmVirtQemuKernel-AARCH64/DEBUG_GCC49/FV/QEMU_EFI.fd
```

Comment:
So, we have `$(ROOT)/build` which contains source files (build.git), and `$(ROOT)/Build` which is the output of `edk2` and `edk2-platforms`? That seems a bit confusing to me. Is there no way to keep EDK2 files under `edk2/build`?

Reply:
That is confusing and a bit annoying, but there seems to be no `O=...` equivalent when building EDK2, not what I've found at least.

Reply:
port it to autotools!

Reply:
Keep `EDK2_BIN` unchanged and set `WORKSPACE=$(EDK2_PATH)` in `common.mk` 
(https://github.com/tianocore/tianocore.github.io/wiki/Multiple_Workspace#environment-variables says "WORKSPACE: location of the Build directory").

Reply:
Ah, that seems to work fine, thanks for looking that up @jforissier . Btw, this requires a change in `fvp.mk` too. I've changed that in the fix I just submitted. Haven't tested it on FVP yet, but I'm going to do that now.

Reply:
Sounds good

Reply:
Unfortunately this doesn't seem to work when building FVP.  With full path it complains that it cannot find the dsc-file and if I remove the full path, like this:
```diff
-               -t "GCC49" -p $(EDK2_PLATFORMS_PATH)/Platform/ARM/VExpressPkg/ArmVExpress-FVP-AArch64.dsc -b RELEASE
+               -t "GCC49" -p Platform/ARM/VExpressPkg/ArmVExpress-FVP-AArch64.dsc -b RELEASE
```
Then it seems to find the dsc-file and compiles some stuff, but ends with:
```bash
GenFds.py...
 : error F003: Output file for RAW section could not be found for Platform/ARM/VExpressPkg/AcpiTables/AcpiTables.inf

#########


build.py...
 : error 7000: Failed to execute command
        GenFds -f /mnt/sshd/devel/optee_projects/fvp/edk2-platforms/Platform/ARM/VExpressPkg/ArmVExpress-FVP-AArch64.fdf --conf=/mnt/sshd/devel/optee_projects/fvp/edk2/Conf -o /mnt/sshd/devel/optee_projects/fvp/edk2/Build/ArmVExpress-FVP-AArch64/RELEASE_GCC49 -t GCC49 -b RELEASE -p /mnt/sshd/devel/optee_projects/fvp/edk2-platforms/Platform/ARM/VExpressPkg/ArmVExpress-FVP-AArch64.dsc -a AARCH64 -D "EFI_SOURCE=/mnt/sshd/devel/optee_projects/fvp/edk2/EdkCompatibilityPkg" -D "EDK_SOURCE=/mnt/sshd/devel/optee_projects/fvp/edk2/EdkCompatibilityPkg" -D "TOOL_CHAIN_TAG=GCC49" -D "TOOLCHAIN=GCC49" -D "TARGET=RELEASE" -D "FAMILY=GCC" -D "WORKSPACE=/mnt/sshd/devel/optee_projects/fvp/edk2" -D "EDK_TOOLS_PATH=/mnt/sshd/devel/optee_projects/fvp/edk2/BaseTools" -D "ARCH=AARCH64" -D "ECP_SOURCE=/mnt/sshd/devel/optee_projects/fvp/edk2/EdkCompatibilityPkg" [/mnt/sshd/devel/optee_projects/fvp/edk2]

- Failed -
Build end time: 10:54:03, Nov.27 2017
Build total time: 00:00:28
```

I suspect this is working on QEMU because we (incorrectly) are doing a shortcut by calling `make` directly in QEMU, but (correctly) using the `build` script when building for FVP. Also, the `*.dsc` file for QEMU lives in `edk2` while the the `dsc` for FVP lives in edk2-platforms. 

I've been testing just a clean edk2/edk2-platforms according to the instructions here:
https://github.com/tianocore/edk2-platforms

Running the https://github.com/tianocore/edk2-platforms#using-uefi-tools-helper-scripts with `$WORKSPACE` set to `edk2` will also make that script fail. So ... I don't really know what to do here.

Reply:
Exporting `WORKSPACE` to `edk2-platforms` instead of `edk2` does seems to work for FVP.